### PR TITLE
Irctc ticket booking with conftest file 

### PR DESCRIPTION
**Pull request by sandeepkrishnagopalappa**
1. created conftest instead of base fixtures.
2.created instances in init instead of creating instances in every irctc pom page and irctc flight page 
3.used driving locators  from objectjson page 
4.created  dictionary  format of data to json data in object creation page 
5.used one method instead of two methods inside test page

### CODE REVIEW

File: Data/dataset_creation.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import json
+from config import *
```
Comment (rohsmiles) [inline Data/dataset_creation.py]: .

File: conftest.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from Library.web_utility import GenericMethod
+from Library.file_library import ReadJson
+from config import *
+from selenium import webdriver
+
+""" This is the conftest file which contains commonly used file in all pages """
+
+options = webdriver.ChromeOptions()
+options.add_argument("--disable-notifications")
+driver = webdriver.Chrome(executable_path, options=options)
```
Comment (rohsmiles) [inline conftest.py]: You can have these code in a fixture and make sure fixture should give you the driver as output.

File: pom/flight_pom.py [hunk 3]
```diff
@@ -1,72 +1,53 @@
+from conftest import *
```
Comment (rohsmiles) [inline pom/flight_pom.py]: Conftest file shouldn't be imported to any file.

File: pom/irctc_pom.py [hunk 4]
```diff
@@ -1,26 +1,19 @@
-from Library.web_utility import GenericMethod
-from Library.file_library import XlData
-from Library.basefixture import *
+from conftest import *
```
Comment (rohsmiles) [inline pom/irctc_pom.py]: Same here, don't import conftest.

File: test/test_reg.py [hunk 5]
```diff
@@ -1,22 +1,20 @@
-"""  This page is the page where pytest detects automatically and
-    through which all the methods or APIs are called through  """
-
-from config import *
+import pytest
 from pom.irctc_pom import Register
 from pom.flight_pom import FlightRegister
-from Library.basefixture import Driver_init
+from conftest import url,driver
```
Comment (rohsmiles) [inline test/test_reg.py]: Same here, don't import conftest file.

File: config.py [hunk 6]
```diff
@@ -6,3 +6,4 @@
 sheet_name= 'object'
 expected_irctc_page_title="FlightTicketBooking Next Generation eTicketing System"
 expected_flight_page_title="Air Ticket Booking | Book Flight Tickets | Cheap Air Fare - FlightTicketBooking Air"
+OBJECT_JSON=r"C:\Users\7000026201\PycharmProjects\FlightTicketBooking\Data\object_data.json"
```
Comment (rohsmiles) [inline config.py]: I cannot run test case on your system always to get the data from the specified path right ??

File: pom/flight_pom.py [hunk 7]
```diff
@@ -1,72 +1,55 @@
...
 
     def station_from_sending_keys(self):
-        reg4 = FlightRegister.variable1_["station_from_sending_keys"]
-        FlightRegister.ObjectGen_.enter_text(self.driver, reg4, values="Chennai")
+        self.ObjectGen_.enter_text(self.driver, self.variable1_["station_from_sending_keys"], values="Chennai")
 
     def station_from_click(self):
-        reg5 = FlightRegister.variable1_["station_from_click"]
-        FlightRegister.ObjectGen_.click_on_element(self.driver, reg5)
+        self.ObjectGen_.click_on_element(self.driver, self.variable1_["station_from_click"])
```
Comment (rohsmiles) [inline pom/flight_pom.py:23]: Make these 2 method into one method.
It should take argument of the name station name.
Your not suppose to hardcode anything.

File: pom/flight_pom.py [hunk 8]
```diff
@@ -1,72 +1,55 @@
...
 
     def station_to_sending_keys(self):
-        reg6 = FlightRegister.variable1_["station_to_sending_keys"]
-        FlightRegister.ObjectGen_.enter_text(self.driver, reg6, values="Mumbai")
+        self.ObjectGen_.enter_text(self.driver, self.variable1_["station_to_sending_keys"], values="Mumbai")
 
     def station_to_click(self):
-        reg7 = FlightRegister.variable1_["station_to_click"]
-        FlightRegister.ObjectGen_.click_on_element(self.driver, reg7)
+        self.ObjectGen_.click_on_element(self.driver, self.variable1_["station_to_click"])
```
Comment (rohsmiles) [inline pom/flight_pom.py:29]: Make these 2 method into 1 method.
It should take argument of the name station name.
Your not suppose to hardcode anything.

File: pom/flight_pom.py [hunk 9]
```diff
@@ -1,72 +1,55 @@
...
 
     def return_date_click(self):
-        reg9 = FlightRegister.variable1_["return_date_click"]
-        FlightRegister.ObjectGen_.click_on_element(self.driver, reg9)
+       self.ObjectGen_.click_on_element(self.driver, self.variable1_["return_date_click"])
 
     def apirl_month_select(self):
-        reg10 = FlightRegister.variable1_["apirl_month_select"]
-        FlightRegister.ObjectGen_.click_on_element(self.driver, reg10)
+        self.ObjectGen_.click_on_element(self.driver, self.variable1_["apirl_month_select"])
```
Comment (rohsmiles) [inline pom/flight_pom.py:38]: Make start_date_select() and apirl_month_select() into one method, which takes date as input parameter.

File: pom/flight_pom.py [hunk 10]
```diff
@@ -1,72 +1,55 @@
...
 
     def return_date_select(self):
-        reg11 = FlightRegister.variable1_["return_date_select"]
-        FlightRegister.ObjectGen_.click_on_element(self.driver, reg11)
+        self.ObjectGen_.click_on_element(self.driver, self.variable1_["return_date_select"])
```
Comment (rohsmiles) [inline pom/flight_pom.py:41]: Make return_date_click() and return_date_select() as 1 method, which takes date as input parameter.

File: pom/flight_pom.py [hunk 11]
```diff
@@ -1,72 +1,55 @@
...
-        FlightRegister.ObjectGen_.click_on_element(self.driver, reg12)
-        return FlightRegister.ObjectGen_.validate_element_display(self.driver, FlightRegister.variable1_["book"])
+        self.ObjectGen_.click_on_element(self.driver, self.variable1_["search"])
+        return self.ObjectGen_.validate_element_display(self.driver,self.variable1_["book"])
 
     def book(self):
-        reg13 = FlightRegister.variable1_["book"]
-        FlightRegister.ObjectGen_.click_on_element(self.driver, reg13)
-        return FlightRegister.ObjectGen_.validate_element_display(self.driver, FlightRegister.variable1_["gst_pop_up"])
+        self.ObjectGen_.click_on_element(self.driver, self.variable1_["book"])
```
Comment (rohsmiles) [inline pom/flight_pom.py:48]: Method name is not clear, I didn't understood what this method does.

File: test/test_reg.py [hunk 12]
```diff
@@ -1,21 +1,19 @@
-"""  This page is the page where pytest detects automatically and
-    through which all the methods or APIs are called through  """
-
-from config import *
+import pytest
 from pom.irctc_pom import Register
 from pom.flight_pom import FlightRegister
-from Library.basefixture import Driver_init
+from config import url
```
Comment (rohsmiles) [inline test/test_reg.py:4]: All these import shouldn't present in test case.
Everything should come through conftest file.

File: conftest.py (hunk 2 above)
Comment (sandeepkrishnagopalappa) [reply conftest.py]: I have used it inside fixture and used request object to over come it

File: pom/flight_pom.py (hunk 3 above)
Comment (sandeepkrishnagopalappa) [reply pom/flight_pom.py]: removed this things sir

File: pom/irctc_pom.py (hunk 4 above)
Comment (sandeepkrishnagopalappa) [reply pom/irctc_pom.py]: removed

File: test/test_reg.py (hunk 5 above)
Comment (sandeepkrishnagopalappa) [reply test/test_reg.py]: removed

File: config.py (hunk 6 above)
Comment (sandeepkrishnagopalappa) [reply config.py]: removed absolute path and used relative path

File: pom/flight_pom.py (hunk 7 above)
Comment (sandeepkrishnagopalappa) [reply pom/flight_pom.py:23]: merged this two methods

File: pom/flight_pom.py (hunk 8 above)
Comment (sandeepkrishnagopalappa) [reply pom/flight_pom.py:29]: merged this two

File: pom/flight_pom.py (hunk 9 above)
Comment (sandeepkrishnagopalappa) [reply pom/flight_pom.py:38]: this i merged with below two ,return date click and return date select

File: pom/flight_pom.py (hunk 10 above)
Comment (sandeepkrishnagopalappa) [reply pom/flight_pom.py:41]: merged both

File: pom/flight_pom.py (hunk 11 above)
Comment (sandeepkrishnagopalappa) [reply pom/flight_pom.py:48]: this is book button click . i have change the name

**Comment by sandeepkrishnagopalappa** [comment]
merged three commits

**Comment by sandeepkrishnagopalappa** [comment]
still few things to commit

**Pull request merged**